js/client.js

```diff
@@ -1563,13 +1608,11 @@
 			if (Config.server.id !== 'showdown') return;
 			var autojoins = [];
 			var autojoinCount = 0;
-			for (var i in this.rooms) {
-				if (!this.rooms[i]) continue;
-				if (this.rooms[i].type !== 'chat' || i === 'lobby') {
-					continue;
-				}
-				autojoins.push(this.rooms[i].id.indexOf('-') >= 0 ? this.rooms[i].id : (this.rooms[i].title || this.rooms[i].id));
-				if (i === 'staff' || i === 'upperstaff') continue;
+			for (var i = 0; i < this.sideRoomList.length; i++) {
```

Comment:
I mean that you should write your code assuming any kind of room can be in any list.